### DISCUSSION TITLE
Server-only function called from Telefunc gives warning about client-side import

### DISCUSSION THREAD

**Discussion by bravius** [vikejs/vike#1768] [Discussion] [answered]
When calling a function defined in a server-only module from a Telefunc function (using the .server/.shared/.client file environment scheme) I get the following warning:

```
7:40:10 AM [vike][Warning] Server-only module /src/db.server.ts (https://vike.dev/file-env) imported on the client-side by /src/rpc/admin.telefunc.ts (building your app for production will be prevented and an error will be thrown).
```

Technically it's correct that the function is imported on the client-side, but Telefunc transforms it to execute on the server-side.

Considering this Vike + Telefunc case, should it actually be throwing this warning and preventing build?

**Comment by brillout** [comment]
That's a bug, can you publish a reproduction?

**Reply by bravius** [reply]
Minimal reproduction [here](https://github.com/bravius/vike-warning-repro)

```
git clone https://github.com/bravius/vike-warning-repro.git
cd vike-warning-repro
yarn
yarn dev
```

Result:

```
10:26:04 PM [vike][Warning] Server-only module /lib/testlib.server.ts (https://vike.dev/file-env) imported on the client-side by /pages/Page.telefunc.ts (building your app for production will be prevented and an error will be thrown).
```

Note: the warning doesn't seem to occur in javascript projects, just typescript

**Reply by brillout** [reply, marked as answer]
[Fix](https://github.com/vikejs/vike/commit/2f53249bf8c4bb7c3a9da261487ad5d73ca465e7) pre-released as `0.4.181-commit-2f53249`.
